tools/resource-gen/main.go

```diff
@@ -19,6 +19,126 @@ import (
 	_ "github.com/kumahq/kuma/api/mesh/v1alpha1"
 )
 
+// NativeApuTemplate for creating a Kuma resource.
+var NativeApiTemplate = template.Must(template.New("resource").Parse(`
+// Generated by tools/resource-gen
+// Run "make generate" to update this file.
+
+{{ $pkg := printf "%s_proto" .Package }}
+{{ $tk := "` + "`" + `" }}
+
+// nolint:whitespace
+package v1alpha1
+
+import (
+	"github.com/golang/protobuf/proto"
+
+	{{ $pkg }} "github.com/kumahq/kuma/api/{{ .Package }}/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/pkg/model"
+	"github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/pkg/registry"
+
+)
+
+{{range .Resources}}
+{{- if not .SkipNativeGenerate }}
+
+// +kubebuilder:object:root=true
+type {{.ResourceType}} struct {
+	metav1.TypeMeta   {{ $tk }}json:",inline"{{ $tk }}
+	metav1.ObjectMeta {{ $tk }}json:"metadata,omitempty"{{ $tk }}
+
+	Mesh   string                        {{ $tk }}json:"mesh,omitempty"{{ $tk }}
+{{- if eq .ResourceType "DataplaneInsight" }}
+	Status   *{{ $pkg }}.{{.ProtoType}} {{ $tk }}json:"status,omitempty"{{ $tk }}
+{{- else}}
+	Spec   *{{ $pkg }}.{{.ProtoType}} {{ $tk }}json:"spec,omitempty"{{ $tk }}
```

Comment:
What does crd-gen do with this?

Reply:
I don't know, but for now preserving existing behavior. Leaving this hack in place until we determine if it should just be changed. Opening an issue on that.